src/Builders/ComponentParameterBuilder.cs

```diff
@@ -0,0 +1,351 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using EC = Microsoft.AspNetCore.Components.EventCallback;
+
+namespace Bunit
+{
+    /// <summary>
+    /// A builder to set a value for strongly typed ComponentParameters.
+    /// </summary>
+    /// <typeparam name="TComponent">The type of component under test to add the parameters</typeparam>
+    public sealed class ComponentParameterBuilder<TComponent> where TComponent : class, IComponent
+    {
+        private const string ParameterNameChildContent = "ChildContent";
+        private readonly List<ComponentParameter> _componentParameters = new List<ComponentParameter>();
+
+        /// <summary>
+        /// Add an unnamed attribute value for a component under test.
+        /// </summary>
+        /// <param name="key">The value to set for an unnamed cascading parameter</param>
+        /// <param name="value">The value to set for an unnamed cascading parameter</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add(string key, object value)

```

Comment:
```suggestion
        public ComponentParameterBuilder<TComponent> AddUnmatched(string key, object value)
```
Isnt it better to explicitly name this `AddUnmatched`, to make it clear that this should only be used if there is an `CaptureUnmatchedValues` parameter in the component?

Reply:
ok; i will rename